app/src/main/java/com/asfoundation/wallet/ui/balance/BalanceFragmentPresenter.kt

```diff
@@ -64,4 +68,44 @@ class BalanceFragmentPresenter(private val view: BalanceFragmentView,
         .subscribe())
   }
 
+  private fun updateAppcBalance(balance: TokenBalance) {
+    var tokenBalance = "-1"
+    var fiatBalance = "-1"
+    if (balance.token.amount.compareTo(BigDecimal("-1")) == 1) {
+      tokenBalance =
+          formatter.formatCurrency(balance.token.amount.toDouble(), WalletCurrency.APPCOINS)
+      fiatBalance = formatter.formatCurrency(balance.fiat)
+    }
+    view.updateTokenValue(tokenBalance, fiatBalance, WalletCurrency.APPCOINS, balance.fiat.symbol)
+  }
+
+  private fun updateCreditsBalance(balance: TokenBalance) {
+    var tokenBalance = "-1"
+    var fiatBalance = "-1"
+    if (balance.token.amount.compareTo(BigDecimal("-1")) == 1) {
+      tokenBalance =
+          formatter.formatCurrency(balance.token.amount.toDouble(), WalletCurrency.CREDITS)
+      fiatBalance = formatter.formatCurrency(balance.fiat)
+    }
+    view.updateTokenValue(tokenBalance, fiatBalance, WalletCurrency.CREDITS, balance.fiat.symbol)
+  }
+
+  private fun updateEthereumBalance(balance: TokenBalance) {
+    var tokenBalance = "-1"
+    var fiatBalance = "-1"
+    if (balance.token.amount.compareTo(BigDecimal("-1")) == 1) {
+      tokenBalance =
+          formatter.formatCurrency(balance.token.amount.toDouble(), WalletCurrency.ETHEREUM)
+      fiatBalance = formatter.formatCurrency(balance.fiat)
+    }
+    view.updateTokenValue(tokenBalance, fiatBalance, WalletCurrency.ETHEREUM, balance.fiat.symbol)
+  }
+
+  private fun updateOverallBalance(balance: FiatValue) {
+    var overallBalance = "-1"
+    if (balance.amount.compareTo(BigDecimal("-1")) == 1) {
+      overallBalance = formatter.formatCurrency(balance)
+    }
+    view.updateOverallBalance(overallBalance, balance.currency, balance.symbol)
+  }
```

Comment:
These methods seems to have a lot of duplicated code. can we merge them an perfom actions bases on WalletCurrency?